scripts/engine/V3/generic-engine-deployer.ts

```diff
@@ -0,0 +1,541 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+import hre from "hardhat";
+import { ethers } from "hardhat";
+import { Contract } from "ethers";
+import path from "path";
+import fs from "fs";
+var util = require("util");
+// hide nuisance logs about event overloading
+import { Logger } from "@ethersproject/logger";
+Logger.setLogLevel(Logger.levels.ERROR);
+import prompt from "prompt";
+
+import {
+  DELEGATION_REGISTRY_ADDRESSES,
+  KNOWN_ENGINE_REGISTRIES,
+} from "../../util/constants";
+import { tryVerify } from "../../util/verification";
+// image bucket creation
+import { createPBABBucket } from "../../util/aws_s3";
+// delay to avoid issues with reorgs and tx failures
+import { delay } from "../../util/utils";
+const EXTRA_DELAY_BETWEEN_TX = 1000; // ms
+const MANUAL_GAS_LIMIT = 500000; // gas
+var log_stdout = process.stdout;
+
+// These are the core contracts that may be deployed by this script.
+const SUPPORTED_CORE_CONTRACTS = ["GenArt721CoreV3_Engine"];
+
+/**
+ * This script was created to deploy the V3 core Engine contracts,
+ * including the associated minter suite, to the Ethereum goerli testnet.
+ * It is intended to document the deployment process and provide a
+ * reference for the steps required to deploy the V3 core contract suite.
+ * IMPORTANT: This deploys a basic randomizer, which may be changed after
+ * deployment by the configured superAdmin.
+ */
+async function main() {
+  prompt.start();
+  const deplomentConfigFile = (
+    await prompt.get<{ from: string }>(["deployment config file"])
+  )["deployment config file"];
+  // dynamically import input deployment configuration details
+  let deployDetailsArray;
+  const fullImportPath = path.join("../../../", deplomentConfigFile);
```

Comment:
I'm not sure if it makes sense to do this vs. just expecting an absolute path – my inclination would be towards the former but would love to better understand your thinking here @ryley-o

Reply:
good call, updated!